components/common/Portal.tsx

```diff
@@ -0,0 +1,29 @@
+import { FC, PropsWithChildren, useEffect, useId, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+interface Props extends PropsWithChildren<unknown> {
+  portalId?: string;
+}
+const Portal: FC<Props> = ({ portalId, children }) => {
+  const [element, setElement] = useState<HTMLElement | null>(null);
+  const id = useId();
+  const elementId = portalId ?? id;
```

Comment:
p2; 혹시 모르니 useId의 리턴값을 바로 사용하기보단 prefix를 붙여서 id를 부여하는게 더 안전할 것 같아요.
```suggestion
  const elementId = portalId ?? `portal-${id}`;
```